src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -239,11 +271,11 @@ along with the reference to the initiating ClassLoader (if the initiating ClassL
                 // and just retry until it succeeds, but in the end we set the interrupt flag
                 // back on to let the interrupt in the next earliest occasion.
 
-                while (true) {
+                int tries = 0;
+                while (tries < MAX_RETRIES) {
```

Comment:
You really don't like `for` loops, do you? ;P

Reply:
I'm not sure why I used a while loop here instead of a for loop. Maybe I thought the while was more expressive of the intended early exit? Maybe because it was already a while? It probably would be better to convert to for loops.

Reply:
I was just thinking back to the `for (foo = bar; foo != null; foo = foo.bar)` loop, and now I saw a normal for loop not using the syntax, so I thought you might have a thing :P